backend/gatekeeper/views.py

```diff
@@ -11,65 +11,79 @@
 from django.urls import reverse
 from django.views.decorators.csrf import csrf_exempt
 from guardian.shortcuts import assign_perm
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
-from gatekeeper.models import Datastream, Thing
-from gatekeeper.utils import parse_sta_url
+from .models import Datastream, Thing
+from .utils import parse_sta_url
 
 
-@csrf_exempt
-def index(request, path):
-    """This is a quick and dirty proof of concept for proxying the SensorThings API queries to the real STA server.
+class Gatekeeper(APIView):
```

Comment:
How does APIView handle OPTIONS? I think it should be forwarded to the STS too.

Reply:
Added